tx_prelude/src/lib.rs

```diff
@@ -361,3 +363,36 @@ impl TxEnv for Ctx {
 pub fn tx_ibc_execute() {
     unsafe { namada_tx_ibc_execute() }
 }
+
+/// Verify section signatures against the given list of keys
+pub fn verify_signatures_of_pks(
+    ctx: &Ctx,
+    tx: &Tx,
+    pks: Vec<common::PublicKey>,
+) -> EnvResult<bool> {
+    let max_signatures_per_transaction =
```

Comment:
What if the `max_signatures_per_transaction` is lower than the number of validator keys at initialization?

Reply:
As we don't have support for validator multisig, it would only be an issue if the source of the tx is a mutisig with threshold > `max_signatures_per_transaction - 5`. We use `15` as the default which should be sufficient for either case

Reply:
ah, I was wrong on this - we do have multisig validator support, but with the default this would still allow up to 10 multisig participants which should be sufficient